src/test/string-enum-wrapper.spec.ts

```diff
@@ -0,0 +1,350 @@
+        },
+      },
+      required: ["wrapper"],
+    });
+
+    // Check that StringChoiceModel was also included in the schema
+    const stringChoiceModel = jsonDoc.components?.schemas?.StringChoiceModel;
+    expect(stringChoiceModel).toBeDefined();
+    expect(stringChoiceModel).toMatchObject({
+      type: "object",
+      properties: {
+        choice: {
+          type: "string",
+        },
+      },
+      required: ["choice"],
+    });
+
+    // Validate successful request with any string
+    const successResponse = await fastify.inject({
+      method: "POST",
+      path: "/string-choice",
+      payload: { wrapper: { choice: "hello" } },
+    });
+
+    expect(successResponse.statusCode).toBe(200);
+    expect(JSON.parse(successResponse.body)).toEqual({ pong: true });
+  });
+
+  test("correctly validates StringEnum objects inside nested objects", async () => {
+    const fastify = Fastify(fastifyOpts);
+    await fastify.register(oas3Plugin, { ...pluginOpts });
+
+    await fastify.register(async (fastify: FastifyInstance) => {
+      fastify.post("/asdf", {
+        schema: {
+          body: AsdfChoiceWrapperModel,
+          response: {
+            200: PingResponse,
+          },
+        },
+        handler: async (req, reply) => {
+          const body = req.body as Static<typeof AsdfChoiceWrapperModel>;
+          return { pong: body.wrapper.choice === "a" };
+        },
+      });
+    });
+    await fastify.ready();
+
+    // Validate OpenAPI document
+    const jsonResponse = await fastify.inject({
+      method: "GET",
+      path: "/openapi.json",
+    });
+
+    const jsonDoc = JSON.parse(jsonResponse.body);
+    const schema = jsonDoc.components?.schemas?.AsdfChoiceWrapperModel;
+    expect(schema).toBeDefined();
+    expect(schema).toMatchObject({
+      type: "object",
+      properties: {
+        wrapper: {
+          $ref: "#/components/schemas/AsdfChoiceModel",
+        },
+      },
+      required: ["wrapper"],
+    });
+
+    // Check that AsdfChoiceModel was also included in the schema
+    const asdfChoiceModel = jsonDoc.components?.schemas?.AsdfChoiceModel;
+    expect(asdfChoiceModel).toBeDefined();
+    expect(asdfChoiceModel).toMatchObject({
+      type: "object",
+      properties: {
+        choice: {
+          type: "string",
+          enum: ["a", "s", "d", "f"],
+        },
+      },
+      required: ["choice"],
+    });
+
+    // Validate successful request
+    const successResponse = await fastify.inject({
+      method: "POST",
+      path: "/asdf",
+      payload: { wrapper: { choice: "a" } },
+    });
+
+    expect(successResponse.statusCode).toBe(200);
+    expect(JSON.parse(successResponse.body)).toEqual({ pong: true });
+
+    // Should fail validation - 'z' is not in enum
+    const errResponse = await fastify.inject({
+      method: "POST",
+      path: "/asdf",
+      payload: { wrapper: { choice: "z" } },
+    });
+
+    console.log(errResponse.body);
```

Comment:
This request returns 500 and the log message prints:

```
{"statusCode":500,"error":"Internal Server Error","message":"Cannot read properties of undefined (reading 'choice')"}
```

If you get rid of the wrapper layer or change the StringEnum to a String, there's no error.